src/server/string_family.cc

```diff
@@ -417,28 +416,41 @@ SinkReplyBuilder::MGetResponse OpMGet(bool fetch_mcflag, bool fetch_mcver, const
   SinkReplyBuilder::MGetResponse response(keys.Size());
   absl::InlinedVector<DbSlice::ConstIterator, 32> iters(keys.Size());
 
+  // First, fetch all iterators and count total size ahead
   size_t total_size = 0;
   unsigned index = 0;
   for (string_view key : keys) {
     auto it_res = db_slice.FindReadOnly(t->GetDbContext(), key, OBJ_STRING);
-    auto& dest = iters[index++];
-    if (!it_res)
-      continue;
-    dest = *it_res;
-    total_size += (*it_res)->second.Size();
+    if (auto& dest = iters[index++]; it_res) {
+      dest = *it_res;
+      total_size += (*it_res)->second.Size();
+    }
   }
 
+  // Allocate enough for all values
   response.storage_list = SinkReplyBuilder::AllocMGetStorage(total_size);
   char* next = response.storage_list->data;
 
   for (size_t i = 0; i < iters.size(); ++i) {
     auto it = iters[i];
-    if (it.is_done())
+    if (it.is_done())  // might have been evicted due to reading other keys
```

Comment:
can you explain this comment?

Reply:
Ok, might be wrong. I remember we had a bug around here with invalidation, there are just no comments about it 🤷🏻‍♂️